### PR TITLE
[Mono] Pass SIMD type variables using SIMD registers

### LINKED ISSUE #74583: [mono] Re-enable passing Vector128 arguments in SIMD registers
Due to mixing LLVM and non-LLVM code (see https://github.com/dotnet/runtime/issues/73454) we disabled the work done for https://github.com/dotnet/runtime/pull/68991 to pass arguments in SIMD registers.  Once the JIT (and interp?) can also use the SIMD registers, we should re-enable that code.

https://github.com/dotnet/runtime/pull/74582

### PR DESCRIPTION
**Pull request by fanyang-mono**
Re-enabled passing SIMD type variables using SIMD registers for LLVM backend.
Added support for passing SIMD type variables using SIMD registers for mini JIT on arm64.

Fixes #74583
Contributes to #60068

### CODE REVIEW

File: src/mono/mono/mini/mini-arm64.c [hunk 1]
```diff
@@ -3109,6 +3110,15 @@ mono_arch_emit_outarg_vt (MonoCompile *cfg, MonoInst *ins, MonoInst *src)
 			MONO_EMIT_NEW_STORE_MEMBASE (cfg, OP_STOREI8_MEMBASE_REG, ARMREG_SP, ainfo->offset + (i * 8), load->dreg);
 		}
 		break;
+	case ArgInSIMDReg:
+		MONO_INST_NEW (cfg, load, OP_LOADX_MEMBASE);
+		load->dreg = mono_alloc_ireg (cfg);
```
Comment (jandupej) [inline src/mono/mono/mini/mini-arm64.c:3115]: Shouldn't this be a `alloc_xreg` (even though internally it seems to do the same thing)

File: src/mono/mono/mini/mini-arm64.c [hunk 2]
```diff
@@ -1529,15 +1529,26 @@ is_hfa (MonoType *t, int *out_nfields, int *out_esize, int *field_offsets)
...
 	int i, size, align_size, nregs, nfields, esize;
 	int field_offsets [16];
 	guint32 align;
+	MonoClass *klass;
 
+	klass = mono_class_from_mono_type_internal (t);
 	size = mini_type_stack_size_full (t, &align, cinfo->pinvoke);
 	align_size = ALIGN_TO (size, 8);
 
+	if (m_class_is_simd_type (klass) && !cinfo->pinvoke && !is_return && cinfo->fr < FP_PARAM_REGS) {
```
Comment (jandupej) [inline src/mono/mono/mini/mini-arm64.c:1543]: I'm not sure if we are following the standard aarch64 calling convention, shouldn't we also return vector types in vector registers?

**Review by jandupej — APPROVED** [review]

File: src/mono/mono/mini/mini-arm64.c (hunk 2 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/mini-arm64.c:1543]: Eventually, yes. That requires more work.

File: src/mono/mono/mini/mini-arm64.c [hunk 3]
```diff
@@ -2806,21 +2818,6 @@ mono_arch_get_llvm_call_info (MonoCompile *cfg, MonoMethodSignature *sig)
 			break;
 		}
 		case ArgVtypeInIRegs:
-#if 0
```
Comment (vargaz) [inline src/mono/mono/mini/mini-arm64.c:2809]: This could be removed.

Comment (fanyang-mono) [reply src/mono/mono/mini/mini-arm64.c:2809]: It has been removed.

**Comment by fanyang-mono** [comment]
Failures on `Build ios-arm64 Release AllSubsets_Mono` were not related to this PR.

**Pull request merged**